src/http-adapter.js

```diff
@@ -76,32 +80,39 @@ class HttpAdapter {
    * @returns {Promise} Promise for destroy.
    */
   destroy (link) {
-    return this._ajax('DELETE', link)
+    return this._http(METHOD.DELETE, link)
   }
 
-  _ajax (type, url, data) {
-    let headers = {
-      'Accept': 'application/vnd.api+json',
-      'Content-Type': 'application/vnd.api+json'
+  async _http (method = '', url = '', data = {}) {
+    let init = {
+      method,
+      headers: {
+        'Accept': 'application/vnd.api+json',
+        'Content-Type': 'application/vnd.api+json'
+      }
     }
 
-    // Stringify data before any async stuff, just in case it's accidentally a mutable object (e.g.
-    // some instrumented Vue data)
-    if (data && ['PATCH', 'POST'].indexOf(type) > -1) {
-      data = JSON.stringify(data)
+    url = new URL(url, location.origin)
+
+    if (method.includes(METHOD.GET)) {
+      // serialize query params
+      Object.keys(data).forEach(key => url.searchParams.append(key, data[key]))
```

Comment:
We can use `Object.entries` here to make things a bit clearer.

Reply:
We'll need to include the polyfill for IE and Safari, too. Maybe use `querystring` instead, since we're already using it in `groveco/grove`...?

Reply:
I can implement those details

Reply:
core-js will take care of `Object.entries`. `searchParams` is part of the `URL` polyfill and would require an upgrade t version 3.